lib/kafka_rest_client.js

```diff
@@ -100,16 +103,29 @@ KafkaRestClient.prototype.getProduceInterval = function getProduceInterval(retry
 KafkaRestClient.prototype.scheduleTopicMapRefresh = function scheduleTopicMapRefresh() {
     var self = this;
     if (self.proxyRefreshTime > 0) {
-        // console.log('Trying to schedule topic refresh job every: ' +
-        // self.proxyRefreshTime / 1000 + ' seconds.');
         /* eslint-disable no-undef,block-scoped-var */
         self.refreshIntervalId = setInterval(function refreshTopics() {
+            self.scheduleConnect(0);
+        }, self.proxyRefreshTime);
+        /* eslint-enable no-undef,block-scoped-var */
+    }
+};
+
+// Reconnect to rest proxy if scheduleTopicMapRefresh fails
+KafkaRestClient.prototype.scheduleConnect = function scheduleConnect(retry) {
+    var self = this;
+    if (retry < connectWaitTimeMs.length) {
+        /* eslint-disable no-undef,block-scoped-var */
+        setTimeout(function reconnect() {
             self.discoverTopics(function refreshTopicsCallback(err, isDone) {
                 if (err === null && isDone) {
                     self.enable = true;
+                } else if (retry < connectWaitTimeMs.length - 1) {
+                    self.enable = false;
```

Comment:
how about set self.connecting to true in this branch? 
Otherwise, the subsequent data requests will get exception till self.enable may become true later, before that, we are still reconnecting.